topic5/movies/src/components/AddComponent.jsx

```diff
@@ -0,0 +1,90 @@
+import React,{Component} from 'react';
+import Movie from './Movie.jsx';
+
+
+class AddComponent extends Component {
+	constructor(props){
+		super(props);
+		this.state = {
+            movieText: '',
+            movies: [],
+		};
+	}
+
+updateMovieText(movieText){
+    this.setState({movieText: movieText.target.value})
+}
+
+addMovie(){
+    if(this.state.movieText === ''){return}
+
+    let moviesArr = this.state.movies;
+    moviesArr.push(this.state.movieText);
+    this.setState({movieText: ''})
+    this.textInput.focus();
+}
+
+handleKeyPress = (event) => {//enables to add when pressing enter on keyboard
+    if(event.key === 'Enter'){
+    let moviesArr = this.state.movies;
+    moviesArr.push(this.state.movieText);
+    this.setState({movieText: ''})
+    }
+}
+
+deleteMovie(index) {
+    const movies = this.state.movies;
+    const newMovies = [
+        ...movies.slice(0, index),
+        ...movies.slice(index + 1)
+    ];
+    this.setState({
+        movies: newMovies
+    });
+}
+
+
+editMovie(index,value){
+    const movies = this.state.movies;
```

Comment:
this is not right: here you're mutating state. What you should do is const movies  = { ...this.state.movies }.

This way you create a new array. After editing what  you need, you can set state again.